crates/db_connection_pool/src/snowflakepool.rs

```diff
@@ -124,6 +123,86 @@ impl SnowflakeConnectionPool {
     }
 }
 
+fn init_snowflake_api(
+    params: &Arc<Option<HashMap<String, String>>>,
+    secret: &Option<Secret>,
+) -> Result<SnowflakeApi> {
+    let username = get_param(params, secret, "username")
+        .context(MissingRequiredSecretSnafu { name: "username" })?;
+    let account = get_param(params, secret, "account")
+        .context(MissingRequiredSecretSnafu { name: "account" })?;
+    // account identifier can be in <orgname.account_name> format but API requires it as <orgname-account_name>
+    let account = account.replace('.', "-");
+
+    let warehouse = get_param(params, secret, "snowflake_warehouse");
+    let role = get_param(params, secret, "snowflake_role");
+    let auth_type = get_param(params, secret, "snowflake_auth_type")
+        .unwrap_or_else(|| "snowflake".to_string())
+        .to_lowercase();
+
+    if auth_type == "snowflake" {
+        // default username/password auth
+        let password = get_param(params, secret, "password")
+            .context(MissingRequiredSecretSnafu { name: "password" })?;
+        let api = SnowflakeApi::with_password_auth(
+            &account,
+            warehouse.as_deref(),
+            None,
+            None,
+            &username,
+            role.as_deref(),
+            &password,
+        )
+        .context(UnableToConnectSnafu)?;
+
+        return Ok(api);
```

Comment:
I would refactor this and the block after `if auth_type == "keypair"` into separate methods.

Reply:
Refactored t0 have separate methods - `init_snowflake_api_with_password_auth` and `init_snowflake_api_with_keypair_auth`